auth_service/auth_adapter/core/auth.py

```diff
@@ -18,121 +18,207 @@
 
 import json
 import logging
+from datetime import datetime
 from typing import Any, Optional
 
-from jwcrypto import jwk, jws
+from jwcrypto import jwk, jwt
 
 from ...config import CONFIG, Config
+from ...deps import Depends, UserDao
+from ...user_management.models.dto import StatusChange, User, UserStatus
 
-__all__ = ["exchange_token", "signing_keys"]
+__all__ = ["exchange_token", "jwt_config"]
 
 log = logging.getLogger(__name__)
 
 
-class SigningKeys:
-    """A container for external and internal signing keys."""
+class JWTConfig:
+    """A container for the JWT related configuration."""
 
     external_jwks: Optional[jwk.JWKSet] = None  # the external public key set
     internal_jwk: Optional[jwk.JWK] = None  # the internal key pair
+    external_algs: Optional[list[str]] = None  # allowed external signing algorithms
+    check_claims: dict[str, Any] = {  # claims that shall be verified
+        "iat": None,
+        "exp": None,
+        "jti": None,
+        "sub": None,
+        "name": None,
+        "email": None,
+        "token_class": "access_token",
+    }
+    # the claims that are copied from the external to the internal token
+    copied_claims = ("name", "email", "iat", "exp")
+    # the key under which the subject is copied from the external token
+    copy_sub_as = "ls_id"
 
     def __init__(self, config: Config = CONFIG) -> None:
-        """Load all the signing keys from the configuration."""
+        """Load the JWT related configuration parameters."""
+
         external_keys = config.auth_ext_keys
-        try:
-            if not external_keys:
-                raise ValueError("No external keys configured")
-            self.external_jwks = jwk.JWKSet.from_json(external_keys)
-        except Exception as exc:  # pylint:disable=broad-except
-            # do not throw an error so that the auth adapter can still run
-            # even though it will not be able to validate external tokens
-            log.error("Error in external signing keys: %s", exc)
+        if external_keys:
+            try:
+                self.external_jwks = jwk.JWKSet.from_json(external_keys)
+            except Exception as exc:  # pylint:disable=broad-except
+                # do not throw an error so that the auth adapter can still run
+                # even though it will not be able to validate external tokens
+                log.error("Cannot parse external signing keys: %s", exc)
+        else:
+            log.error("No external signing keys configured.")
```

Comment:
Shouldn't we raise and exception here, too?

Reply:
PMO

Reply:
Changed this now to fail early when the keys are missing.

The idea was to allow the service to run even without keys so that public requests can be handled and let it fail only when there are authentication requests.

Reply:
But as discussed, it may be better to detect problems early.

Reply:
Changed in d72a549.